cmd/kubelet/kubelet.go

```diff
@@ -80,6 +80,12 @@ func main() {
 		die(err)
 	}
 
+	// re-parse the command-line flags on top of the returned configuration
+	// we layer flags over file-based and remote configuration to
+	// preserve backwards compatibility across binary upgrades
+	// see issue #56171 for more details
+	pflag.Parse()
```

Comment:
I didn't expect to call the global Parse... do we not have access to the flag set and args to re-call?

Reply:
Might require plumbing a flagset to the Bind functions, rather than binding into the global flagset

Reply:
Don't some of the flags bound to config vars behave differently on the first Set call? Reparsing into the same flagset doesn't seem like it would work right.

Reply:
Unfortunately there are third party libs that globally register flags instead of allowing flagset injection. 
I realized you can temporarily substitute values with a noop Set operation to avoid unintentionally calling Set twice for the global flags, details in the re-do PR: #56995

Reply:
We shouldn't be using or depending on those for our kubelet flag parsing